### PR TITLE
ocamlPackages.tar: 2.0.1 → 2.2.2

### PR DESCRIPTION
**Pull request by vbgl**
###### Description of changes

Fixes & improvements: https://github.com/mirage/ocaml-tar/blob/v2.2.2/CHANGES.md

cc maintainer @ulrikstrid

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/ocaml-modules/tar/default.nix [hunk 1]
```diff
@@ -1,40 +1,38 @@
...
 
   buildInputs = [
-    ppx_tools
+    ppx_cstruct
   ];
 
   doCheck = true;
 
   meta = {
     description = "Decode and encode tar format files from Unix";
```
Comment (ulrikstrid) [inline pkgs/development/ocaml-modules/tar/default.nix:34]: While we're at it we can fix my bad description
```suggestion
    description = "Decode and encode tar format files in pure OCaml";
```

**Review by ulrikstrid — APPROVED** [review]
Just a small thing, but this looks good

**Comment by vbgl** [comment]
Please squash the two commits when merging (to whoever takes care of merging this PR).

**Comment by tomodachi94** [comment]
> Please squash the two commits when merging (to whoever takes care of merging this PR).

If you have access to the CLI, you can do that by running `git rebase -i HEAD~3`. Change the 5307f198f012cce75935a5fb32a401afe10bc229 commit's command to `squash`, then write the file and close your text editor. After that, run `git push -f` to force update your branch.

**Pull request merged**